src/monospace/index.ts

```diff
@@ -0,0 +1,115 @@
+import fetch from "node-fetch";
+import { GoogleAuth } from "google-auth-library";
+
+import * as apiv2 from "../apiv2";
+import { FirebaseError } from "../error";
+
+import type { GetInitFirebaseResponse, InitFirebaseResponse } from "./interfaces";
+
+const MONOSPACE_DAEMON_PORT = process.env.MONOSPACE_DAEMON_PORT;
+
+/**
+ * Integrate Firebase Plugin with Monospace’s service Account Authentication
+ */
+export async function setupMonospace(client: GoogleAuth, project?: string): Promise<void> {
+  const accessToken = await client.getAccessToken();
+
+  if (!accessToken) throw new Error("Failed to get access token");
+
+  apiv2.setAccessToken(accessToken);
+
+  const initFirebaseResponse = await initFirebase(project);
+
+  if (initFirebaseResponse.success === false) {
+    // TODO: handle errors case by case if we ever have anything different than NOT_INITIALIZED?
+    throw new Error(String(initFirebaseResponse.error));
+  }
+
+  const { rid } = initFirebaseResponse;
+
+  // Poll for response from the user
+  const authorizedProject = await pollAuthorizedProject(rid);
+
+  if (project === authorizedProject) return;
+
+  // TODO: create a .firebaserc in user’s folder like this:
```

Comment:
Can this return the project instead? The VS Code extension has logic to manage reading/writing `.firebaserc` and can take it from here.

Reply:
It's returning the project now if `options.isVSCE` is true